src/Psr/CacheItemPool/CacheItem.php

```diff
@@ -31,45 +34,32 @@ final class CacheItem implements CacheItemInterface
 
     /**
      * True if the cache item lookup resulted in a cache hit or if they item is deferred or successfully saved
-     *
-     * @var bool
      */
-    private $isHit = false;
+    private bool $isHit;
 
     /**
      * Timestamp item will expire at if expiresAt() called, null otherwise
-     *
-     * @var int|null
      */
-    private $expiration;
+    private ?int $expiration = null;
 
-    /**
-     * Seconds after item is stored it will expire at if expiresAfter() called, null otherwise
-     *
-     * @var int|null
-     */
-    private $ttl;
-
-    /** @var DateTimeZone */
-    private $utc;
+    private Clock $clock;
 
     /**
-     * @param string $key
      * @param mixed $value
-     * @param bool $isHit
      */
-    public function __construct($key, $value, $isHit)
+    public function __construct(string $key, $value, bool $isHit, ?Clock $clock = null)
     {
         $this->key   = $key;
         $this->value = $isHit ? $value : null;
         $this->isHit = $isHit;
-        $this->utc   = new DateTimeZone('UTC');
+        $clock     ??= SystemClock::fromSystemTimezone();
+        $this->clock = $clock;
```

Comment:
Having the clock on a `CacheItem` seems weird to me: this makes a `CacheItem` a mutable/stateful object that depends on global state :thinking:

Reply:
`getTtl()` makes it kinda required to have this design.